src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -856,7 +857,37 @@ pass::EliminateIdentity::EliminateIdentity() {
         if (!identity) {
             return false;
         }
-        return replace_output_update_name(identity->output(0), identity->input_value(0));
+
+        auto identity_out = identity->output(0);
+        auto replacement = identity->input_value(0);
+
+        if (!replace_output_update_name(identity_out, replacement)) {
+            const auto& consumers = replacement.get_target_inputs();
+            bool replacement_has_result_consumer =
+                std::any_of(consumers.cbegin(), consumers.cend(), [](const Input<Node>& consumer) {
+                    return ov::is_type<op::v0::Result>(consumer.get_node());
+                });
+            // Edge case from ticket 177222
+            // NPU doesn't support Identity, so until then
+            // we use a Reshape as a temp no-op to preserve both tensors
+            if (replacement_has_result_consumer) {
+                auto shape_of = std::make_shared<ov::op::v3::ShapeOf>(replacement, ov::element::i64);
+                auto reshape = std::make_shared<ov::op::v1::Reshape>(replacement, shape_of, false);
+
+                // Preserve the "Identity" friendly name on the new node, so any
+                // output still sees the same logical node name
+                reshape->set_friendly_name(identity->get_friendly_name());
+
+                // Preserve Identity's tensor names on the new output, so externally
+                // visible names for that path remain stable
+                reshape->output(0).get_tensor().set_names(identity_out.get_tensor().get_names());
```

Comment:
If Identity has tensor names they should be set on previous node output tensor (append).
Or it may be required to append them as Result (model output) names.

It should be covered by some tests.